app/Http/Controllers/Admin/SupplierController.php

```diff
@@ -131,4 +135,82 @@ public function destroy($id)
 
         return abort(config('setting.errors404'));
     }
+
+    public function showProduct($id)
+    {
+        if (Auth::user()->can('importProduct', Supplier::class)) {
+            $supplier = Supplier::findOrFail($id);
+            $products = Product::all();
+
+            return view('admin.suppliers.import_product', compact('supplier', 'products'));
+        }
+    }
+
+    public function showModal($idProduct, $idSupplier)
```

Comment:
```suggestion
    public function showModal($productId, $supplierId)
```
modal này show thông tin gì em? em biết ntn chung chung quá

Reply:
cái này 2 tham số truyền vào ạ. id của product và id của supplier, dùng 2 thông tin này đấy để hiển thị trên action của form modal ( gồm nhập size và số lượng và giá nếu thay đổi) mà cái route nhập hàng của em sử dụng 2 param đầu vào ấy em buộc phải truyền như vậy ạ id của sản phẩm e để trên url, và id của supplier em để trong body khi ajax post ạ.